weather_mv/loader_pipeline/ee_util.py

```diff
@@ -0,0 +1,457 @@
+            properties: metadata properties
+
+        Returns:
+            The Task ID of the EE ingestion request
+        """
+        manifest = {
+            'name': asset_id,
+            'tilesets': [{
+                'id': 'data_tile',
+                'crs': "epsg:4326",
+                'sources': [{
+                    'uris': [out_path]
+                }]
+            }],
+            'bands': [{
+                'id': f"{variable}",
+                'tileset_id': 'data_tile',
+                'tileset_band_index': idx,
+            } for idx, variable in enumerate(variable_list)],
+            'start_time': {
+                'seconds': start_time
+            },
+            'end_time': {
+                'seconds': end_time
+            },
+            'properties': properties
+        }
+
+        logger.info(f"Uploading GeoTiff {out_path} to Asset ID {asset_id}.")
+        task_id = self.start_ingestion_task(manifest)
+        return task_id
+
+    def process(self, grib_data: GribData) -> t.Iterator[str]:
+        """Writes tiff file, upload it to bucket and ingest into earth engine"""
+
+        # Initializing earth engine
+        if not self.dry_run:
+            ee_initialize(use_personal_account=self.use_personal_account)
+
+        file_name = f'{grib_data.name}.tiff'
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            local_tiff_path = os.path.join(temp_dir, file_name)
+            with rasterio.open(local_tiff_path, 'w',
+                               driver='GTiff',
+                               dtype=grib_data.dtype,
+                               width=grib_data.data[0].data.shape[1],
+                               height=grib_data.data[0].data.shape[0],
+                               count=len(grib_data.data),
+                               crs=grib_data.projection,
+                               transform=grib_data.transform,
+                               tiled=True) as f:
+                for i, da in enumerate(grib_data.data):
+                    f.write(da, i+1)
+
+            # Copy local tiff to gcs tiff location
+            storage_client = storage.Client()
+            parsed_tiff_location = urlparse(self.tiff_location)
+            bucket_name = parsed_tiff_location.netloc
+            bucket = storage_client.get_bucket(bucket_name)
+
+            tiff_internal_path = str(parsed_tiff_location.path)
+            tiff_internal_path = tiff_internal_path[1:] if tiff_internal_path.startswith('/') else tiff_internal_path
+            tiff_bucket_path = os.path.join(tiff_internal_path, file_name)
+
+            logger.debug(f'Copying {local_tiff_path} to {tiff_bucket_path}')
+            blob = bucket.blob(tiff_bucket_path)
+            blob.upload_from_filename(local_tiff_path)
+
+            if self.dry_run:
+                # In case of dry run, we'll not push tiff into earth engine.
+                yield tiff_bucket_path
+                return
```

Comment:
A dry run should involve no data movement whatsoever. This does a lot of it (both downloading and uploading grib/tiffs).